### PR TITLE
UX fixes in discussion reply editing

### PR DESCRIPTION
**Pull request by hypeJunction**
Show spinner while loading the form
Bring form to viewport once display is toggled (in long discussion replies it's hard to locate it)

### CODE REVIEW

File: mod/discussions/views/default/discussion/discussion.js [hunk 1]
```diff
@@ -30,13 +30,21 @@ elgg.discussion.Reply.prototype = {
...
+		$form.slideDown('medium').data('hidden', 0);
+		$('body').animate({
+			scrollTop: $form.offset().top
+		}, 500);
 	},
 
 	loadForm: function () {
 		var that = this;
-
+		var spinner = require('elgg/spinner');
```
Comment (mrclay) [inline mod/discussions/views/default/discussion/discussion.js]: You can't use synchronous require outside of a module. Might work due to race condition but it won't be reliable.

File: mod/discussions/views/default/discussion/discussion.js [hunk 2]
```diff
@@ -30,13 +30,21 @@ elgg.discussion.Reply.prototype = {
 
 	showForm: function () {
 		this.getForm().slideDown('medium').data('hidden', 0);
```
Comment (mrclay) [inline mod/discussions/views/default/discussion/discussion.js]: This line should go away, right?

Comment (hypeJunction) [inline mod/discussions/views/default/discussion/discussion.js]: Yes

**Comment by hypeJunction** [comment]
I am rewriting discussion JS to AMD. Will do this for 2.2.

**Pull request closed without merging**